### ISSUE TITLE
Multiple tabs in visual panel mode get overwritten by loaded layout when switching

### ISSUE THREAD

**Issue by grindingpeon** [radareorg/radare2#23267]
## Environment

```sh
2024-09-01T18:06:30 CEST
radare2 5.9.2 0 @ linux-x86-64
birth: git.5.9.2 2024-08-01__08:35:56
options: gpl release -O1 cs:5 cl:2 meson
Linux x86_64
```

## Description

When using visual panel mode and loading my default layout with `v default` I try to create another tab.
But when switching back to the first one the second gets overwritten with the loaded layout.
I.e. I cannot have different layouts in the different tabs while they all get overwritten to the one loaded.
Is this on purpose or how can I keep the different layouts in the different tabs?

**Comment by grindingpeon** [comment]
I guess a video can show exactly what I mean:

https://github.com/user-attachments/assets/8745ed86-3f1b-4857-9f74-f5e64d64ad56

**Comment by trufae** [comment]
Looks like a bug to me. Each tab should have a different layout

**Comment by grindingpeon** [comment]
Is there a way I can help solving the bug? :thinking: 
I am not the most proficient in C and debugging.

**Comment by trufae** [comment]
Sorry i had other priorities and being busy, sure it will be great if you can try to fix the bug by yourself. Can you join the Discord o rTelegram chats? the best way to debug in panels is by using `eprintf()+sleep(1)`. but feel free to investigate by reading the code and making few changes to unsderstand the behaviour .

can you give me the keys you press so i can repro? because by using it the way i do it seems to work fine for me

**Comment by grindingpeon** [comment]
Yes, will join at some point, don't have time to work on it atm.

This is my layout named default stored inside `~/.local/share/radare2/r2panels`:
```
{"Title":"Disassembly","Cmd":"pd","x":0,"y":1,"w":99,"h":28},{"Title":"Hexdump","Cmd":"xc $r*16","x":0,"y":28,"w":87,"h":19},{"Title":"RegisterRefs","Cmd":"drr","x": 98,"y":1,"w":76,"h":28},{"Title":"Stack","Cmd":"pxr@r:SP","x":86,"y":28,"w":88,"h":19}
```

To reproduce I type:

```
r2 -AA /bin/ls
# inside radare2
v default
v
# open new tab
tt
# switch
t2
# change layout
"j<enter>
# switch back to first tab
t1
# switch back to second tab
t2
# the layout has been changed to to the default :/
```

**Comment by trufae** [comment]
Using your r2panels json i get this message all the time

<img width="1188" alt="Screenshot 2024-11-19 at 12 08 32 PM" src="https://github.com/user-attachments/assets/ef1bf3c3-144f-4955-8a82-c0ab72e7d306">

i think there are some more issues related to layout saving/restoring. not critical, but anoying, ill find some time after the big major refactoring abi breaking seasson that starts today. feel free to jump in when you have some spare time to chk it because i can't handle it right now. thanks!

**Comment by trufae** [comment]
anyone wants to give it a try and improve/fix whatever is wrong here?

**Comment by trufae** [comment]
shuold be fixed in https://github.com/radareorg/radare2/commit/1c3fedfc1ebd8ff2cae377ba57d5c6f7edee9149
